cli/src/lib.rs

```diff
@@ -24,14 +32,79 @@ struct Args {
     /// SQL file to execute
     #[clap(short, long, value_parser)]
     execute: Option<PathBuf>,
+
+    /// PATH to dump whole database
+    #[clap(short, long, value_parser)]
+    dump: Option<PathBuf>,
 }
 
-pub fn run() {
+pub fn run() -> Result<(), Box<dyn Error>> {
     let args = Args::parse();
 
     if let Some(path) = args.path {
         let path = path.as_path().to_str().expect("wrong path");
 
+        if let Some(dump) = args.dump {
+            let file = File::create(dump)?;
+            let storage = SledStorage::new(path).expect("failed to load sled-storage");
+            block_on(async {
+                let (storage, _) = storage.begin(true).await.map_err(|(_, error)| error)?;
+                let schemas = storage.fetch_all_schemas().await?;
+                stream::iter(&schemas)
+                    .for_each(|schema| async {
+                        writeln!(&file, "{}", schema.clone().to_ddl()).unwrap();
+
+                        storage
+                            .scan_data(&schema.table_name)
+                            .await
+                            .map(stream::iter)
+                            .unwrap()
+                            .map_ok(|(_, row)| row)
+                            .try_chunks(100)
+                            .try_for_each(|rows| {
+                                let exprs_list = rows
+                                    .into_iter()
+                                    .map(|Row(values)| {
+                                        values
+                                            .into_iter()
+                                            .map(|value| {
+                                                Expr::Literal(AstLiteral::try_from(value).unwrap())
+                                            })
+                                            .collect::<Vec<_>>()
+                                    })
+                                    .collect::<Vec<_>>();
+
+                                let insert_statement = Statement::Insert {
+                                    table_name: schema.table_name.clone(),
+                                    columns: Vec::new(),
+                                    source: gluesql_core::ast::Query {
+                                        body: SetExpr::Values(Values(exprs_list)),
+                                        order_by: Vec::new(),
+                                        limit: None,
+                                        offset: None,
+                                    },
+                                }
+                                .to_sql();
+
+                                writeln!(&file, "{}", insert_statement)
+                                    .map_err(ready)
+                                    .unwrap();
```

Comment:
So hard to merge TryChunkError + I/O Error + gluesqlError here but eventually we will just unwrap at main..